app/lib/actions/observable_action.rb

```diff
@@ -0,0 +1,37 @@
+module Actions
+  module ObservableAction
+    def self.included(base)
+      base.include ::Foreman::Observable
+      base.execution_plan_hooks.use :emit_event, :on => :stopped
+
+      def base.event_name
+        # actions/remote_execution/run_host_job_stopped
+        ::Foreman::Observable.event_name_for(to_s.underscore + '_stopped')
+      end
+    end
```

Comment:
I understand that this means a task was stopped which can mean it was a success or it is in stopped state until someone unpause it. I was wondering what events are available and if we should also emit some of the other ones?

Reply:
So far it emits the event only when the task goes into stopped state (stopped for good, no pausing, resuming or anything). It doesn't necessarily mean it was a success, it just means the task will not run anymore. If it was a success or not can be read from the payload.

We could emit an event each time a task goes into a different state (one of pending, scheduled, planning, planned, running, paused, stopped) or on success (state stopped, result success) or on failure (state stopped, result error). I can add whatever you feel is useful, I just didn't want to overdo it from the start.

Reply:
I wonder if makes sense to emit _processed or _finished when it was a success. This way all webhook users will need to do if-statement in the ERB, it could be cleaner if we'd do this for them here. I am not sure how much work is this to implement, e.g. for ActiveRecord it's possible to pass options in the form of `:if => { # condition }`.

Otherwise let's just merge.

Reply:
I understand, I just think that requiring users to do processing of the payload everytime when they just want to subscribe to "success" tasks would be nice.

Would you mind doing `_finished` or something like that and then `_stopped` for all the others?

Reply:
Went with `${action}_succeeded` for now